### PR TITLE
refactor observability components

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

This pr is to decouple es and fluent bit components from logging. Now es and fluent bit are common components. Installer will choose to deploy them if any of logging, events and auditing has been enabled.

I have tested this PR with the following cases:

- Enable events only
- Enable auditing first, then logging
- Enable both events and logging + External ES
- Install with non-default es setting (elkPrefix, logMaxAge)

### CODE REVIEW

File: roles/ks-auditing/tasks/fluentbit-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    src: "fluentbit-operator"
+    dest: "{{ kubesphere_dir }}/"
+
+
+- name: ks-auditing | Creating manifests
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "{{ kubesphere_dir }}/fluentbit-operator/{{ item.file }}"
+  with_items:
+    - { name: custom-output-elasticsearch, file: custom-output-elasticsearch.yaml }
```
Comment (benjaminhuo) [inline roles/ks-auditing/tasks/fluentbit-operator.yaml:12]: item.name is not used?
audit input CR needn't to be changed??

File: roles/ks-events/tasks/fluentbit-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    src: "fluentbit-operator"
+    dest: "{{ kubesphere_dir }}/"
+
+
+- name: ks-events | Creating manifests
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "{{ kubesphere_dir }}/fluentbit-operator/{{ item.file }}"
+  with_items:
+    - { name: custom-output-elasticsearch, file: custom-output-elasticsearch.yaml }
```
Comment (benjaminhuo) [inline roles/ks-events/tasks/fluentbit-operator.yaml:12]: same as audit:
item.name is not used?
events input CR needn't to be changed??

File: roles/ks-logging/tasks/main.yaml [hunk 3]
```diff
@@ -1,188 +1,20 @@
...
-    - { type: crd, name: fluentbits.logging.kubesphere.io }
-  when:
-    - kubesphere_version.stdout.split('.')[0] == "2"
-
-
-- name: ks-logging | Prepare fluentbit operator setup
-  shell: >
-    {{ bin_dir }}/kubectl apply -f {{ kubesphere_dir }}/logging/fluentbit-operator/init
-
+    - { name: custom-output-elasticsearch, file: custom-output-elasticsearch.yaml }
```
Comment (benjaminhuo) [inline roles/ks-logging/tasks/main.yaml:13]: item.name is not used?
logging input CR needn't to be changed?

File: roles/ks-auditing/tasks/fluentbit-operator.yaml (hunk 1 above)
Comment (huanggze) [reply roles/ks-auditing/tasks/fluentbit-operator.yaml:12]: `item.name` isn't used, code style across installer though.

Both `input` and `filter` are static, only `output` might be various in line with different es setup.

File: roles/ks-events/tasks/fluentbit-operator.yaml (hunk 2 above)
Comment (junotx) [reply roles/ks-events/tasks/fluentbit-operator.yaml:12]: @huanggze The output paths of `custom-output-elasticsearch.yaml` for logging, events, auditing are the same, and may cause conflicts or overwrites. For events, the `dest` can be changed to `{{ kubesphere_dir }}/ks-events/fluentbit-operator/{{ item.file }}`  to differentiate.

Comment (benjaminhuo) [reply roles/ks-events/tasks/fluentbit-operator.yaml:12]: yeah, that makes sense. they're different CR. Or just add a prefix like events-custom-output-elasticsearch.yaml

**Comment by benjaminhuo** [comment]
I'm thinking once one of logging/events/auditing is installed, and then fluentbit is installed and maybe configured, so fluentbit CR might contains settings user may not want to lose. So if user choose to disable any of the logging/events/auditing components, we could simply change fluentbit input CR label enable to false and change it back to true once they're enabled again if the corresponding CR exists or else create a new input CR for that component.

I think ks-installer is still lack of uninstalling process like this for many components.
We may add the function I mentioned above later when we need to considering the entire uninstalling process.

File: roles/ks-events/tasks/fluentbit-operator.yaml (hunk 2 above)
Comment (huanggze) [reply roles/ks-events/tasks/fluentbit-operator.yaml:12]: that's true

Comment (huanggze) [reply roles/ks-events/tasks/fluentbit-operator.yaml]: done, xxx.yaml -> xxx-events.yaml

File: kubesphere-minimal.yaml [hunk 4]
```diff
@@ -25,6 +25,13 @@ data:
       etcdVolumeSize: 20Gi
       openldapVolumeSize: 2Gi
       redisVolumSize: 2Gi
+      es:
```
Comment (benjaminhuo) [inline kubesphere-minimal.yaml:28]: we needn't to add these to kubesphere-complete-setup.yaml?

Comment (huanggze) [reply kubesphere-minimal.yaml:28]: kubesphere-complete-setup.yaml updated.

**Pull request merged**